main.py

```diff
@@ -42,8 +107,47 @@ def api_label(my_path=None):
+        check = os.path.isfile(os.path.join('static/data/', (my_path + '.png')))
+        if check == False:
+            return Response(status=400, response='Image does not exist.')
         with open(os.path.join('static/data/', (my_path + '.json')), "w+") as outfile:
-            json.dump(request.get_json(), outfile, sort_keys=True, indent=4)
+            data = request.get_json()
+            json.dump(data, outfile, sort_keys=True, indent=4)
+            # Find the template for this folder
+            template = fetch_template(os.path.join('static/data/',(my_path[:my_path.rindex('/')] + '/template.json')))
+            if template == None:
+                return Response(status=400, response='Unable to find template.')
+            # Open the template
+            template = open(os.path.join(template, 'template.json'), 'r')
+            # Convert the template json to something validate can understand.
+            template = json.load(template)
+           # Call the validate function from jsonschema to verify the file we were suppsoed to create conforms with the template for the file.
+        new_json = open(os.path.join('static/data/', (my_path + '.json')))
+        new_json = json.load(new_json)
+        # print(new_json[-1])
+        # Generally speaking most errors will be caught at this stage of the game, where if it does not match our schema, it will send a validation error.
+        try:
+            validate(new_json, EXPORT_SCHEMA)
+            # print(v)
+        except:
+            return Response(status=400, response='Json does not match the required format, requires 3 points minimum and 1 label.')
+        # There is a chance that the user will manually input a bad json through the network menu. This will catch it. 
+        try:
+            dict_json = new_json[0]
+        except KeyError:
+            return Response(status=400, response='Desired key not found, is this a valid json label?')
+        # Should it somehow pass all of the above there is still a chance the submitted json has incorrect labels applied to it. An except here catches this.
+        try:
+            if dict_json['label']['class'] not in template['classes']:
+                # print(dict_json['label']['class'])
+                return Response(status=400, response='Selected class not available in template.')
+        except:
+        	# If classes is not found in template then we error out. So this is an exra layer of security that shouldn't be necessary.
+            return Response(status=400, response='Unable to find keyword: classes')
+        # print(new_json)
+        # Call the validate function from jsonscheme to now ensure the json created contains the data desired.
+        if new_json != data:
+            return Response(status=400, response='Generated json does not match the data submitted for export.')
         if UPLOAD_LIMIT is not None:
             CURRENT_UPLOAD_SIZE = CURRENT_UPLOAD_SIZE + 1
```

Comment:
I know this isn't your PR but why do we increment the upload count every time we add a label or make a directory? Shouldn't the upload count be for... umm.... uploads?

Reply:
I considered writing any files to disk as an upload, just so that a bot or a person can't spam files in demo mode

Reply:
Makes sense, variable should probably be FILE_MAX or something but w/e